src/protagonist/Engine/Ingest/IngestExecutor.cs

```diff
@@ -32,35 +36,46 @@ public class IngestExecutor
         var workers = workerBuilder.GetWorkers(asset);
         
         var context = new IngestionContext(asset);
+        var overallStatus = IngestResultStatus.Unknown;
 
         if (!assetIngestorSizeCheck.CustomerHasNoStorageCheck(asset.Customer))
         {
             var preIngestionAssetSize = await assetRepository.GetImageSize(asset.Id, cancellationToken);
             context.WithPreIngestionAssetSize(preIngestionAssetSize);
+            
+            var counts = await storageRepository.GetStorageMetrics(asset.Customer, cancellationToken);
+            
+            if (!counts.CanStoreAssetSize(MinimumAssetSize, 0))
+            {

```

Comment:
Is it cleaner to short circuit here and return? Makes it easier to see at a glance that processing stops if true and there's nothing more to do.

```cs
if (!counts.CanStoreAssetSize(MinimumAssetSize, 0))
{
  logger.LogDebug("Something informative");
  asset.Error = "StoragePolicy size limit exceeded";
  var dbSuccess = await CompleteAssetInDatabase(context, true, cancellationToken);
  return new IngestResult(asset, dbSuccess ? IngestResultStatus.StorageLimitExceeded : IngestResultStatus.Failed);
}
```

`"StoragePolicy size limit exceeded"` is used here and `AssetIngestorSizeCheckBase` - worth adding a const?